### PR TITLE
feat(selection): Add triple-click and selection improvements

### PR DESCRIPTION
**Pull request by 0xBigBoss**
## Summary

This PR adds comprehensive selection improvements to ghostty-web:

- **Triple-click line selection**: Select entire lines with triple-click, matching native terminal behavior
- **Browser-native click detection**: Use `event.detail` for reliable double/triple-click detection instead of custom timing logic
- **Path-friendly word selection**: Expand word characters to include `/`, `.`, `-`, `_`, `~` for selecting file paths (matching native Ghostty)
- **Smart line selection**: Triple-click selects actual text content, not full terminal width
- **Scrollback support**: Line selection works correctly in scrollback buffer
- **Single-cell handling**: Properly handle edge cases with single-cell and single-character selections
- **Semi-transparent overlay**: Use VS Code-style semi-transparent selection overlay instead of inverted colors

## Test plan

- [x] All existing tests pass (328 tests)
- [ ] Manual: Double-click selects words including paths like `/usr/local/bin`
- [ ] Manual: Triple-click selects the line content (not trailing whitespace)
- [ ] Manual: Triple-click in scrollback buffer works correctly
- [ ] Manual: Selection overlay is semi-transparent blue (not inverted)
- [ ] Manual: Single-cell clicks don't leave stray highlights

### CODE REVIEW

**Comment by 0xBigBoss** [comment]
## Additional changes for discussion

We have a few more changes in our fork that enable VS Code webview integration. These are potentially VS Code-specific, so I wanted to gauge interest before submitting a separate PR.

**Compare:** https://github.com/coder/ghostty-web/compare/main...0xBigBoss:ghostty-web:feat/integration-hooks

### Changes included:

1. **`onLinkClick` callback** - Allows the host to handle link clicks (e.g., open in external browser from VS Code webview)

2. **Three-way return semantics for `customKeyEventHandler`**
   - Current: `true` = terminal handles, falsy = default processing
   - Proposed: `true` = terminal handles, `false` = bubble to host, `undefined` = default processing
   - Use case: VS Code needs certain key combos (Cmd+P, Cmd+Shift+P) to bubble up to the host instead of being processed by the terminal

3. **Let Cmd/Meta combos bubble to host** - On macOS, Meta+key combos return early so VS Code can handle them (Quick Open, Command Palette, etc.)

Would these be useful upstream, or are they too VS Code-specific? Happy to submit a separate PR if there's interest.

**Comment by sreya** [comment]
@codex review

File: lib/selection-manager.ts [hunk 1]
```diff
@@ -550,6 +550,20 @@ export class SelectionManager {
...
+        if (
+          this.selectionStart &&
+          this.selectionEnd &&
+          this.selectionStart.col === this.selectionEnd.col &&
+          this.selectionStart.absoluteRow === this.selectionEnd.absoluteRow
+        ) {
+          // Clear same-cell selection from click-without-drag
+          this.selectionStart = null;
+          this.selectionEnd = null;
+          return;
```
[the remark this answers is not in the document]
Comment (sreya) [reply lib/selection-manager.ts]: @0xBigBoss I noticed this as well while trying out your changes

**Comment by sreya** [comment]
@codex review

**Comment by sreya** [comment]
@codex review
